src/main/cms/cms_types.h

```diff
@@ -69,11 +69,11 @@ typedef struct
 
 #define IS_PRINTVALUE(p) ((p)->flags & PRINT_VALUE)
 #define SET_PRINTVALUE(p) { (p)->flags |= PRINT_VALUE; }
-#define CLR_PRINTVALUE(p) { (p)->flags &= ~PRINT_VALUE; }
+#define CLR_PRINTVALUE(p) { (p)->flags &= (uint8_t)~PRINT_VALUE; }
 
 #define IS_PRINTLABEL(p) ((p)->flags & PRINT_LABEL)
 #define SET_PRINTLABEL(p) { (p)->flags |= PRINT_LABEL; }
-#define CLR_PRINTLABEL(p) { (p)->flags &= ~PRINT_LABEL; }
+#define CLR_PRINTLABEL(p) { (p)->flags &= (uint8_t)~PRINT_LABEL; }
```

Comment:
The problem with this cast is that it will give and incorrect value if someone, in future, increases the size of `flags` (say to a `uint16_t`) and omits to change this cast. The solution is to define a type for the flags and cast to that. ie:
```
typedef uint8_t OSD_flags_t;

typedef struct
{
    const char *text;
    const OSD_MenuElement type;
    const CMSEntryFuncPtr func;
    void *data;
    OSD_flags_t flags;
} OSD_Entry;
```
Finally the macro should be replaced by a `static inline` function, ie:
```
static inline void CLR_PRINTVALUE(OSD_Entry *p) { p->flags &= (OSD_flags_t)~PRINT_VALUE; }
```